.github/workflows/publish-api-doc.yml

```diff
@@ -0,0 +1,63 @@
+
+# This generates Doxygen autodocs for the C++ API (for the libyui base lib)
+# and pushes them to the "gh-pages" branch which publishes them
+# to the project's GitHub pages at https://libyui.github.io/libyui/api-doc/
+#
+# See also
+#
+#   https://gist.github.com/shundhammer/ed359db0d9329d4db551528256060d2a
+#
+# GitHub actions syntax:
+#   https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+name: publish-api-doc
+
+# Trigger this when a pull request is merged (which implies pushing to master).
+on:
+  push:
+    branches:
+      # master
+      - main
```

Comment:
Um, should it be really `master`?

Reply:
I tried, and it works well with this. IIRC you can define somewhere what the _main_ branch of a repo is, and I'd guess that GitHub actions choose that one here.

Reply:
Changed to _master_.